app/Console/Components/Npm/FetchTotals.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace App\Console\Components\Npm;
+
+use App\Events\Npm\TotalsFetched;
+use Developmint\NpmStats\NpmStats;
+use GuzzleHttp\Client;
+use Illuminate\Console\Command;
+
+class FetchTotals extends Command
+{
+    protected $signature = 'dashboard:fetch-npm-totals';
+
+    protected $description = 'Fetch totals for all our NPM packages';
+
+    public function handle()
+    {
+        $npmStats = new NpmStats(new Client());
+
+        $packageList = $this->getPackageList();
+
+        /*
+         * First idea was to filter the non-scoped and scoped packages so at least the
+         * scoped packages could've been processed by bulk. This would work, but the
+         * allowed range for bulk is lower than normal, which would lead to wrong,
+         * incomplete or weird data in some cases
+         *
+         */
+
+
+        $totals = $packageList->reduce(function ($carry, $package) use ($npmStats) {
+
+            $carry["daily"] += $npmStats->getStats($package)["downloads"];
+            $carry["monthly"] += $npmStats->getStats($package, NpmStats::LAST_MONTH)["downloads"];
+            $carry["total"] += $npmStats->getStats($package, NpmStats::TOTAL)["downloads"];
+
+            return $carry;
+        }, ["daily" => 0, "monthly" => 0, "total" => 0]);
+
+        event(new TotalsFetched($totals));
+    }
+
+    private function getPackageList()
```

Comment:
@freekmurze Maybe our main site should export a JSON with these, since we already maintain a list there?

Reply:
API endpoint created: https://spatie.be/en/api/packages

Reply:
@manniL could you use our new endpoint, instead of hardcoding the package list?

Reply:
Done @freekmurze ☺️